CMakeLists.txt

```diff
@@ -143,23 +141,54 @@ else ()
   message (WARNING "Boost graph headers not found, will not build 'graphs' backend")
 endif ()
 find_package (Boost REQUIRED COMPONENTS iostreams)
+set (HAVE_LIBBOOST_IOSTREAMS 1)
 # otherwise ordered_map code tries to use boost::get (graph)
 add_definitions ("-DBOOST_NO_ARGUMENT_DEPENDENT_LOOKUP")
+
+# Compile with the Boehm garbage collector (https://github.com/ivmai/bdwgc) if requested.
 if (ENABLE_GC)
-  find_package (LibGc 7.4.2 REQUIRED)
+  # We need to enable position independent code to make the malloc redirect work.
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+  set(enable_cplusplus ON CACHE BOOL "C++ support")
+  # We redirect all malloc calls in the program.
+  set(enable_redirect_malloc ON CACHE BOOL "Redirect malloc and friends to GC routines.")
+  set(install_headers OFF CACHE BOOL "Install header and pkg-config metadata files.")
+  set(enable_docs OFF CACHE BOOL "Build and install documentation.")
+  set(enable_large_config ON CACHE BOOL "Optimize for large heap or root set.")
+  add_definitions("-DREDIRECT_REALLOC=GC_REALLOC")
+  add_definitions("-DREDIRECT_FREE=GC_FREE")
+  if (APPLE)
+    append("-Wno-deprecated-declarations" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    # add_definitions("-DGC_NO_PTHREAD_SIGMASK")
+    # add_definitions("-DNO_MARKER_SPECIAL_SIGMASK")
+  endif()
+  if(BUILD_STATIC_RELEASE)
+    set(enable_threads OFF CACHE BOOL "Support threads.")
```

Comment:
why off?

Reply:
I have not been able to successfully compile it with the thread library.

Reply:
Then we can expect performance regressions in static builds (ones we give to customers :/)

Reply:
Not quite sure, if I remember correctly the Ubuntu packages do not have `THREADS` defined by default.

Reply:
E.g., if you try to use `GC_allow_register_threads` it will complain.

Reply:
https://stackoverflow.com/questions/35116327/when-g-static-link-pthread-cause-segmentation-fault-why

I am running into this problem. Somehow pthread is not linked correctly. I have not found an order to make this work. 
This change is possibly related, which makes things quite confusing. 
https://developers.redhat.com/articles/2021/12/17/why-glibc-234-removed-libpthread#the_developer_view
Dropping `-static-libgcc -static-libstdc++` does not seem to help either.

Reply:
What does this option mean? Is it enabling support for running multi-threaded programs with libGC, or is it enabling parallel runs of garbage collecting threads?